client/scripts/views/pages/view_snapshot_proposal/index.ts

```diff
@@ -179,38 +209,64 @@ const ViewProposalPage: m.Component<{
 }, {
   proposal: SnapshotProposal,
   votes: Vote[],
+  space: any,
+  snapshotProposal: any,
+  totalScore: number,
+  scores: number[]
 }> = {
   oninit: (vnode) => {
     vnode.state.votes = [];
+    vnode.state.totalScore = 0;
+    vnode.state.scores = [];
 
     const snapshotId = vnode.attrs.snapshotId;
     app.snapshot.fetchSnapshotProposals(snapshotId).then(response => {
-      
       const allProposals: SnapshotProposal[] = app.snapshot.proposalStore.getAll();
       vnode.state.proposal = allProposals.filter(proposal => proposal.ipfsHash === vnode.attrs.identifier)[0];
 
-      if (vnode.state.proposal) {
-        const hubUrl = process.env.SNAPSHOT_APP_HUB_URL || 'https://testnet.snapshot.org';
-        $.get(`${hubUrl}/api/${snapshotId}/proposal/${vnode.state.proposal.ipfsHash}`).then((response) => {
-          if (response.status !== 'Success') {
-            var i = 0;
-            let votes: Vote[] = [];
-            for (const key in response) {
-              let vote: Vote = {
-                voterAddress: '',
-                choice: '',
-                timestamp: '',
-              };
-              vote.voterAddress = key,
-              vote.timestamp = response[key].msg.timestamp;
-              vote.choice = response[key].msg.payload.choice === 1 ? 'yes' : 'no';
-              votes.push(vote);
-            }
-            vnode.state.votes = votes;
-            m.redraw();
+      snapshotClient.getSpaces().then(response => {
```

Comment:
why do we do this?

Reply:
This feels like something that should happen at snapshot controller initialization, stored on the class, and then utilized top down, rather than each time a page/view/component loads within snapshot.

Reply:
Correct!